spec/minidoc/timestamps_spec.rb

```diff
@@ -20,5 +22,27 @@ class TimestampsUser < Minidoc
       user.save
       expect(user.created_at).to_not eq user.updated_at
     end
+
+    it "updates updated_at when doc is changed with #set" do
+      user = TimestampsUser.create
+      sleep 0.001
+      user.set(name: "Abby Normal")
+      expect(user.created_at).to_not eq user.updated_at
+    end
+
+    it "updates updated_at when doc is changed with .atomic_set" do
```

Comment:
Originally, I had wanted to implement this logic in the class methods. Current implementations made it impractical to do that, unfortunately, because of how the instance methods update the attrs of the current in-memory record instance.

Implementing both was an option, of course, but then you end up with a lot more of the duplication you referred to elsewhere.

Ultimately, the instance methods are the primary public interface, so this seemed like the right way to go. I think the class ones are maybe even just public because Ruby's method visibility is a bit weird with class methods? If this is a concern, I think there could be a refactor to remove the class methods & make them part of the instance methods (or private instance helper methods), but that's a bag of major-version-changing snakes I didn't want to dig into.